### ISSUE TITLE
Root splat route (routes/$.tsx) is not matched on "/%0A" 

### ISSUE THREAD

**Issue by arthurvi** [remix-run/remix#8635]
### Reproduction

I have created a failing test scenario here: https://github.com/arthurvi/remix-line-feed-bug

Simply make sure you have a root splat route (`/routes/$.tsx`). Then visit `/%0A` and see the root splat route is not matched. Instead an error is thrown and the error boundary is triggered. 

> Error: No route matches URL "/%0A"

### System Info

```shell
System:
    OS: macOS 14.2
    CPU: (12) arm64 Apple M2 Pro
    Memory: 2.49 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.volta/tools/image/node/20.10.0/bin/node
    Yarn: 1.22.21 - ~/.volta/tools/image/node/20.10.0/bin/yarn
    npm: 10.2.3 - ~/.volta/tools/image/node/20.10.0/bin/npm
  Browsers:
    Chrome: 120.0.6099.234
    Edge: 121.0.2277.83
    Safari: 17.2
  npmPackages:
    @remix-run/changelog-github: ^0.0.5 => 0.0.5 
    @remix-run/web-fetch: ^4.4.2 => 4.4.2 
    vite: ^5.0.0 => 5.0.0 
```

### Used Package Manager

npm

### Expected Behavior

The root splat route (`/routes/$.tsx`) is matched when the URL is `/%0A`.

### Actual Behavior

The error boundary is triggered when the URL is `/%0A`.

**Comment by brophdawg11** [comment]
This is a duplicate of https://github.com/remix-run/react-router/discussions/11221.  Please read through the linked proposal for more information and add an upvote if that's a feature you'd like to see - and if it gains enough traction it can be considered for implementation.

tl;dr; React Router does not currently support matching newline characters as part of a splat path - and we're not currently intending to add support since newlines in pathnames are pretty edge case and an odd usage of pathnames in our opinion.  This is something you can add support for via `patch-package` if desired by changing the underlying regex's for splat matching.
